dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java

```diff
@@ -736,13 +1092,100 @@ private String getMarker(LogRecord logRecord) {
     }
 
     /**
+     * check casting to WsLogRecord
+     *
      * @return
      */
-    private WsLogRecord getWsLogRecord(LogRecord logRecord) {
+    //change to public
+    public WsLogRecord getWsLogRecord(LogRecord logRecord) {
         try {
             return (WsLogRecord) logRecord;
         } catch (ClassCastException ex) {
             return null;
         }
     }
+
+    /**
+     * Outputs filteredStream of genData
+     *
+     * @param genData object to filter
+     * @return filtered message of the genData
+     */
+    protected String formatStreamOutput(GenericData genData) {
+        String txt = null;
+        String loglevel = null;
+        KeyValuePair kvp = null;
+
+        ArrayList<Pair> pairs = genData.getPairs();
+        for (Pair p : pairs) {
+
+            if (p instanceof KeyValuePair) {
+
+                kvp = (KeyValuePair) p;
+                if (kvp.getKey().equals("message")) {
+                    //txt = kvp.getValue();
+                    txt = kvp.getStringValue();
+                } else if (kvp.getKey().equals("loglevel")) {
+                    //loglevel = kvp.getValue();
+                    loglevel = kvp.getStringValue();
+                }
+            }
+        }
+
+        String message = filterStackTraces(txt);
+        if (message != null) {
+            if (loglevel.equals("SystemErr")) {
+                message = "[err] " + message;
+            }
+        }
+        return message;
+    }
+
+    private String filterStackTraces(String txt) {
```

Comment:
should modify BaseTraceService.filterStackTraces to be public static (dealing with the instance variables that would need to be passed to it) and use that instead of copying the whole method here.